Compiler/src/verifytrim.jl

```diff
@@ -219,9 +221,17 @@ function verify_codeinstance!(codeinst::CodeInstance, codeinfo::CodeInfo, inspec
                     end
                 elseif Core.finalizer isa ftyp
                     if length(stmt.args) == 3
-                        # TODO: check that calling `args[1](args[2])` is defined before warning
+                        finalizer = argextype(stmt.args[2], codeinfo, sptypes)
+                        obj = argextype(stmt.args[3], codeinfo, sptypes)
+                        atype = argtypes_to_type(Any[finalizer, obj])
+
+                        mi = compileable_specialization_for_call(interp, atype)
+                        if mi !== nothing
+                            ci = get(caches, mi, nothing)
+                            ci isa CodeInstance && continue
+                        end
+
                         error = "unresolved finalizer registered"
-                        warn = true
                     end
                 else
                     error = "unresolved call to builtin"
```

Comment:
Could you move this to the top of the `<: Builtin` code? Looks like we have a few places that would be left with the default message instead ("unresolved call"), which may be mildly annoying

Could you also add a check `if !isconcretetype(ftyp); error = "unknown Builtin"; end`? It is unlikely someone manages to infer as just Builtin or a Union, but upon further reflection, it seems like it would probably be confusing that `ftyp isa` would trigger on a random case statement and end up trying to explain why this unknown function would not make be a valid call to specifically `_apply_iterate`